crates/milli/src/search/facet/facet_distribution.rs

```diff
@@ -385,6 +357,61 @@ impl<'a> FacetDistribution<'a> {
 
         Ok(distribution)
     }
+
+    /// Select a field if it is faceted and in the facets.
+    fn select_field(
+        &self,
+        name: &str,
+        metadata: &Metadata,
+        filterable_attributes_rules: &[FilterableAttributesRule],
+    ) -> bool {
+        match &self.facets {
+            Some(facets) => {
+                // The list of facets provided by the user is a legacy pattern ("dog.age" must be selected with "dog").
+                facets.keys().any(|key| match_field_legacy(key, name) == PatternMatch::Match)
+            }
+            None => metadata.is_faceted(filterable_attributes_rules),
+        }
+    }
+
+    /// Check if the fields in the facets are valid faceted fields.
+    fn check_faceted_fields(
+        &self,
+        fields_ids_map: &FieldIdMapWithMetadata,
+        filterable_attributes_rules: &[FilterableAttributesRule],
+    ) -> Result<()> {
+        let mut invalid_facets = BTreeSet::new();
+        if let Some(facets) = &self.facets {
+            for (field, _) in facets {
+                let is_valid_faceted_field =
+                    fields_ids_map.id_with_metadata(field).map_or(false, |(_, metadata)| {
+                        metadata.is_faceted(filterable_attributes_rules)
```

Comment:
I seem to remember that facet distribution would only work with filterable attributes, **not** distinct, sortable, etc. 

The error message of `fn format_invalid_filter_distribution` explicitly mentions `filterable` attributes.

Can we confirm this behavior on a v1.13 and a v1.12?

Reply:
you're right https://github.com/meilisearch/meilisearch-support/issues/702#issuecomment-2703403394